pkg/segment/test/set_functions.go

```diff
@@ -16,9 +16,17 @@ package segment
 
 // Basic numeric constants that we define because the math package doesn't.
 // TODO(nlacasse): These should be Math.MaxInt64/MinInt64?
+
 const (
 	maxInt = int(^uint(0) >> 1)
-	minInt = -maxInt - 1
+	minInt = (-maxInt - 1) / 2
+	//minInt = -maxInt - 1
+	// This is adjusted to make sure no add overflow would happen in test cases
```

Comment:
It's just to ensure no overflow within the test because no range with super large End will happen in the test case.
Actually I tried to change everything to uint in the test, but that would affect TestAddSequentialAdjacent and TestAddSequentialNonAdjacent as I mentioned in the comment and I felt I'd better not change them so I do this compromising setting.
I'm not sure which way is better. Let me know your suggestions.

Reply:
I'll let @nixprime comment. My preference would be for the segment set to actually handle the overflow. E.g.

if gap.Range().Length(); !(length >= 0) {
   ... handle overflow here? ...
} else {
   ... do the normal stuff ...
}

The length >= 0 will actually just be removed from the template for uint64 cases.

But this could get complicated. My second preference would be to actually eliminate the problem by setting minInt to 0 here and explaining why. (Right now it doesn't actually eliminate the problem, since you can still overflow!)

Reply:
The second solution has a problem as I've mentioned that two other tests will be affected. But I guess it could be fixed. 

The first solution is a little bit confusing to me. Even if we know there's a overflow, what can we do with it?  We can't return a uint if an int is expected. And this is not going to happen in real use case and  only occurs in test case. So I feel we'd better adjust the test case instead.

Reply:
@nixprime What's your opinion on this? 

I'm planning to set the minint here to be zero (Adin's second suggestion) and correct the two testcases not written by me which will be affected.